commit f983729654cd9bfb405ee9757583b48814eee32a
Author: jim <yuanjianmin@espressif.com>
Date:   Wed Nov 15 21:09:47 2023 +0800

    Fix enable dynamic mbedtls will occur heap interruption when server support TLS renegotiation

diff --git a/components/mbedtls/port/dynamic/esp_ssl_cli.c b/components/mbedtls/port/dynamic/esp_ssl_cli.c
index e35d62e24d..64bb643787 100644
--- a/components/mbedtls/port/dynamic/esp_ssl_cli.c
+++ b/components/mbedtls/port/dynamic/esp_ssl_cli.c
@@ -183,6 +183,11 @@ static int manage_resource(mbedtls_ssl_context *ssl, bool add)
         case MBEDTLS_SSL_FLUSH_BUFFERS:
             break;
         case MBEDTLS_SSL_HANDSHAKE_WRAPUP:
+#if defined(MBEDTLS_SSL_RENEGOTIATION)
+            if (add && ssl->MBEDTLS_PRIVATE(renego_status)) {
+                CHECK_OK(esp_mbedtls_add_rx_buffer(ssl));
+            }
+#endif
             break;
         default:
             break;
